app/controllers/provider/admin/cms/templates_controller.rb

```diff
@@ -31,7 +31,18 @@ def update
     @page.build_version if params[:version]
 
     if @page.save
-      success_update
+      msg = publish_or_hide_page_with_message
+      respond_to do |format|
```

Comment:
Trying to remove the double `format.a_format` check, this could be something like:
```ruby
respond_to do |format|
  format.html do
    if @page.save
      flash[:notice] = publish_or_hide_page_with_message
      redirect_to(action: :edit, id: @page.id)
    else
      render :edit 
    end
  end

  format.js do
    if @page.save 
      flash.now[:notice] = publish_or_hide_page_with_message
    end
    
    render template: '/provider/admin/cms/templates/update'
  end
end
```
Not that it's better since there's a double `@page.save` now 😄

Reply:
yes!, there are 4 possible paths with 4 different outcomes, it can'tb e reduced to less AFAIK, so there will always be a repeated inner `if`